### PR TITLE
[v9] Add hint when the user receives an error about an "unknown certificate authority" (#11550)

### PR DESCRIPTION
**Pull request by hatched**
Backports #11550 

* Intercept and update error message when there is a certificate error joining a node.

* Log out error hint and return full wrapped error.

* Updated error message.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Comment by zmb3** [comment]
@hatched I just realized that hard-coding `/var/lib/teleport` in the error message is probably not the best idea, since it will confuse users who have configured a different data directory.

Do you want to fix it in master and leave these backport PRs open so you can pull the fix in without needing a second set of backports?

File: lib/service/connect.go [hunk 1]
```diff
@@ -160,6 +160,15 @@ func (process *TeleportProcess) connect(role types.SystemRole) (conn *Connector,
...
 		clt, err := process.newClient(process.Config.AuthServers, identity)
 		if err != nil {
+			// In the event that a user is attempting to connect a machine to
+			// a different cluster it will give a cryptic warning about an
+			// unknown certificate authority. Unfortunately we cannot intercept
+			// this error as it comes from the http package before a request is
+			// made. So provide a more user friendly error as a hint of what
+			// they can do to resolve the issue.
+			if strings.Contains(err.Error(), "certificate signed by unknown authority") {
+				process.log.Error("Was this node already registered to a different cluster? To join this node to a new cluster, remove `/var/lib/teleport` and try again")
```
Comment (hatched) [inline lib/service/connect.go:170]: @zmb3 what do you think of 
```suggestion
				process.log.Error("Was this node already registered to a different cluster? To join this node to a new cluster remove its data directory and try again (default location: `/var/lib/teleport`).")
```

Comment (zmb3) [reply lib/service/connect.go:170]: You have access to the config, so `process.Config.DataDir` will give you the exact directory.

Comment (hatched) [reply lib/service/connect.go:170]: Brilliant thanks!

**Review by Tener — APPROVED** [review]

**Pull request merged**